File: PolicyPlus/PolicyStructures.vb
' These structures hold information on the behavior of policies and policy elements
Public Class PolicyRegistryList
    Public OnValue As PolicyRegistryValue
    Public OnValueList As PolicyRegistrySingleList
    Public OffValue As PolicyRegistryValue
    Public OffValueList As PolicyRegistrySingleList
End Class
Public Class PolicyRegistrySingleList
    Public DefaultRegistryKey As String
    Public AffectedValues As List(Of PolicyRegistryListEntry)
End Class
Public Class PolicyRegistryValue ' <value>
    Public RegistryType As PolicyRegistryValueType
    Public StringValue As String
    Public NumberValue As UInteger
End Class
Public Class PolicyRegistryListEntry ' <item>
    Public RegistryValue As String
    Public RegistryKey As String
    Public Value As PolicyRegistryValue
End Class
Public Enum PolicyRegistryValueType
    Delete
    Numeric
    Text
End Enum
Public MustInherit Class PolicyElement
    Public ID As String
    Public ClientExtension As String
    Public RegistryKey As String
    Public RegistryValue As String
    Public ElementType As String
End Class
Public Class DecimalPolicyElement ' <decimal>
    Inherits PolicyElement
    Public Required As Boolean
    Public Minimum As UInteger
    Public Maximum As UInteger = UInteger.MaxValue
    Public StoreAsText As Boolean
    Public NoOverwrite As Boolean
End Class
Public Class BooleanPolicyElement ' <boolean>
    Inherits PolicyElement
    Public AffectedRegistry As PolicyRegistryList
End Class
Public Class TextPolicyElement ' <text>
    Inherits PolicyElement
    Public Required As Boolean
    Public MaxLength As Integer
    Public RegExpandSz As Boolean
    Public NoOverwrite As Boolean
End Class
Public Class ListPolicyElement ' <list>
    Inherits PolicyElement
    Public HasPrefix As Boolean
    Public NoPurgeOthers As Boolean
    Public RegExpandSz As Boolean
    Public UserProvidesNames As Boolean
End Class
Public Class EnumPolicyElement ' <enum>
    Inherits PolicyElement
    Public Required As Boolean
    Public Items As List(Of EnumPolicyElementItem)
End Class
Public Class EnumPolicyElementItem ' <item>
    Public DisplayCode As String
    Public Value As PolicyRegistryValue
    Public ValueList As PolicyRegistrySingleList ' <valueList>
End Class
Public Class MultiTextPolicyElement ' <multiText>
    Inherits PolicyElement
    ' This is undocumented, so it's unknown whether there can be other options for it
End Class

File: PolicyPlus/PolicyStructures.cs
using System.Collections.Generic;

namespace PolicyPlus
{
    // These structures hold information on the behavior of policies and policy elements
    public class PolicyRegistryList
    {
        public PolicyRegistryValue OnValue;
        public PolicyRegistrySingleList OnValueList;
        public PolicyRegistryValue OffValue;
        public PolicyRegistrySingleList OffValueList;
    }

    public class PolicyRegistrySingleList
    {
        public string DefaultRegistryKey;
        public List<PolicyRegistryListEntry> AffectedValues;
    }

    public class PolicyRegistryValue // <value>
    {
        public PolicyRegistryValueType RegistryType;
        public string StringValue;
        public uint NumberValue;
    }

    public class PolicyRegistryListEntry // <item>
    {
        public string RegistryValue;
        public string RegistryKey;
        public PolicyRegistryValue Value;
    }

    public enum PolicyRegistryValueType
    {
        Delete,
        Numeric,
        Text
    }

    public abstract class PolicyElement
    {
        public string ID;
        public string ClientExtension;
        public string RegistryKey;
        public string RegistryValue;
        public string ElementType;
    }
    public class DecimalPolicyElement : PolicyElement // <decimal>
    {
        public bool Required;
        public uint Minimum;
        public uint Maximum = uint.MaxValue;
        public bool StoreAsText;
        public bool NoOverwrite;
    }
    public class BooleanPolicyElement : PolicyElement // <boolean>
    {
        public PolicyRegistryList AffectedRegistry;
    }
    public class TextPolicyElement : PolicyElement // <text>
    {
        public bool Required;
        public int MaxLength;
        public bool RegExpandSz;
        public bool NoOverwrite;
    }
    public class ListPolicyElement : PolicyElement // <list>
    {
        public bool HasPrefix;
        public bool NoPurgeOthers;
        public bool RegExpandSz;
        public bool UserProvidesNames;
    }
    public class EnumPolicyElement : PolicyElement // <enum>
    {
        public bool Required;
        public List<EnumPolicyElementItem> Items;
    }

    public class EnumPolicyElementItem // <item>
    {
        public string DisplayCode;
        public PolicyRegistryValue Value;
        public PolicyRegistrySingleList ValueList; // <valueList>
    }
    public class MultiTextPolicyElement : PolicyElement // <multiText>
    {
        // This is undocumented, so it's unknown whether there can be other options for it
    }
}